jobs/influxdb/templates/bin/influxdb_setup.erb

```diff
@@ -71,10 +71,14 @@ INFLUXDB_HTTP_PORT=<%= p('influxdb.http.port') %>
 ADMIN_USERNAME="<%= admin_username %>"
 ADMIN_PASSWORD_FILE="/var/vcap/store/influxdb/admin_password_file"
 if [ ! -f "$ADMIN_PASSWORD_FILE" ]; then
-  # generate the admin password
+  # generate the admin password and store it in /var/vcap/store
   openssl rand -hex 16 > "$ADMIN_PASSWORD_FILE"
 fi
-ADMIN_PASSWORD=$(cat "$ADMIN_PASSWORD_FILE")
+if [ $(stat -c %U "$ADMIN_PASSWORD_FILE") != "root" ]; then # TODO: merge with the previous if
+  sudo chown root:root "$ADMIN_PASSWORD_FILE"
+  sudo chmod 600 "$ADMIN_PASSWORD_FILE"
+fi
```

Comment:
the separate `if` is required to migrate existing deployments. once all deployments have migrated the two `if`s can be merged.